engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceProcessor.java

```diff
@@ -213,6 +226,36 @@ private boolean hasMultiInstanceScope(final ExecutableFlowElement flowElement) {
     }
   }
 
+  private Either<Rejection, ?> validateTargetsSupportedElementType(
+      final ExecutableProcess process,
+      final ArrayProperty<ProcessInstanceCreationStartInstruction> startInstructions) {
+
+    return startInstructions.stream()
+        .map(
+            instruction ->
+                Tuple.of(
+                    instruction.getElementId(),
+                    process.getElementById(instruction.getElementIdBuffer()).getElementType()))
```

Comment:
:wrench: We could use a `record` instead of the generic `Tuple`.

It is a minor difference. In general, I would prefer a `record` because the getters have more meaningful names than `getLeft()`/`getRight()`.